Add docstrings and docs pages for `RequiredDataValidator` and `DataValidator` (#470)
IAMconsortium/nomenclature: docs/user_guide/data-validation.rst

.. _data-validation:

.. currentmodule:: nomenclature

Data validation
===============

> The **nomenclature** package allows users to validate IAMC data in two ways.

Reviewer: Write docs future-proof
```suggestion
The **nomenclature** package allows users to validate IAMC data in several ways.
```

> 

Reviewer: Please add a table of content for the two methods.

For this, validation requirements and criteria can be specified in YAML configuration
files.

Required data validation
------------------------

**Required data validation** checks if certain models, variables, regions and/or 
periods of time are covered in the datapoints.

For this, a configuration file specifies the model(s) and dimension(s) expected
in the dataset. These are ``variable``, ``region`` and/or ``year``.

> Alternatively, instead of using ``variable``, it's possible to declare measurands,

Reviewer: ```suggestion
Alternatively, instead of using ``variable``, it is possible to declare measurands,
```

> which can also specify units.

Reviewer: ```suggestion
which jointly specify variables and units.
```

.. code:: yaml

  description: Required variables for running MAGICC
  model: model_a
  required_data:
    - measurand:
        Emissions|CO2:
          unit: Mt CO2/yr
      region: World
      year: [2020, 2030, 2040, 2050]

In the example above, for *model_a*, the dataset must include datapoints of the
variable *Emissions|CO2* (measured in *Mt CO2/yr*), in the region *World*, for the
years 2020, 2030, 2040 and 2050.

Data validation
---------------

> **Data validation** checks if data values are within reasonable ranges.

Reviewer: ```suggestion
**Data validation** checks if data values are within specified ranges.
```
